aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -0,0 +1,500 @@
+        self.states.lock().unwrap().remove(state_key);
+    }
+
+    fn apply_write_set(&self, write_set: WriteSet) {
+        let mut states = self.states.lock().unwrap();
+
+        for (state_key, write_op) in write_set {
+            match write_op.as_state_value() {
+                None => {
+                    states.remove(&state_key);
+                },
+                Some(state_val) => {
+                    states.insert(state_key, Some(state_val));
+                },
+            }
+        }
+    }
+
+    #[allow(dead_code)]
+    fn read_resource<T: MoveResource>(&self, addr: &AccountAddress) -> T {
+        let data_blob = self
+            .get_state_value_bytes(
+                &StateKey::resource_typed::<T>(addr).expect("failed to create StateKey"),
+            )
+            .expect("account must exist in data store")
+            .unwrap_or_else(|| panic!("Can't fetch {} resource for {}", T::STRUCT_NAME, addr));
+
+        bcs::from_bytes(&data_blob).expect("failed to deserialize resource")
+    }
+}
+
+impl<'a, S> TStateView for SimulationStateView<'a, S>
+where
+    S: StateView,
+{
+    type Key = StateKey;
+
+    fn get_state_value(&self, state_key: &Self::Key) -> StateStoreResult<Option<StateValue>> {
+        if let Some(res) = self.states.lock().unwrap().get(state_key) {
+            return Ok(res.clone());
+        }
+        self.remote.get_state_value(state_key)
+    }
+
+    fn get_usage(&self) -> StateStoreResult<StateStorageUsage> {
+        panic!("not supported")
+    }
+
+    fn as_in_memory_state_view(&self) -> InMemoryStateView {
+        panic!("not supported")
+    }
+}
+
+/***************************************************************************************************
+ * Move Patches
+ *
+ **************************************************************************************************/
+static MODULE_ID_CREATE_SIGNER: Lazy<ModuleId> = Lazy::new(|| {
+    ModuleId::new(
+        AccountAddress::ONE,
+        Identifier::new("create_signer").unwrap(),
+    )
+});
+
+static MODULE_ID_APTOS_GOVERNANCE: Lazy<ModuleId> = Lazy::new(|| {
+    ModuleId::new(
+        AccountAddress::ONE,
+        Identifier::new("aptos_governance").unwrap(),
+    )
+});
+
+/// Makes `aptos_framework::create_signer::create_signer` a public function so that
+/// it can be used freely in the simulation.
+fn patch_create_signer(state_view: &SimulationStateView<impl StateView>) -> Result<()> {
+    let resolver = state_view.as_move_resolver();
+    let blob = resolver
+        .get_module(&MODULE_ID_CREATE_SIGNER)?
+        .ok_or_else(|| anyhow!("module create_signer does not exist"))?;
+
+    let mut m = CompiledModule::deserialize(&blob)?;
+    let func_def = find_function_def_by_name(&mut m, IdentStr::new("create_signer").unwrap())
+        .ok_or_else(|| anyhow!("failed to locate `fun create_signer`"))?;
+    func_def.visibility = Visibility::Public;
+
+    move_bytecode_verifier::verify_module(&m)?;
```

Comment:
Do we want to run metadata passes (from AptosVM) here as well?

Reply:
No, I'm calling the verifier just to some basic sanity check, making sure that the manually-crated bytecode is correct. Will add a more detailed error explanation to make the intent clearer.